---
title: "Integration_site_circos_plots"
date: "1/8/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r message=FALSE, warning=FALSE}
library(circlize) # Circular Visualization
library(readxl) # Read Excel Files
library(dplyr) # A Grammar of Data Manipulation
```


```{r}
# read in manually curated Manta integration site calls: ALL INTEGRATION SITES PER SAMPLE; NO PRESELECTION
manta.calls <- read_xlsx("./raw_data/Integration_Sites_merged.xlsx")

# filter for high confidence Manta call set
#manta.calls <- manta.calls %>% filter(Contig != "IMPRECISE" & BND_depth >= 3 & Mate_BND_depth >=3) 

# define which chromosomes to plot
chromosome.list <- c("chr1",
                     "chr2",
                     "chr3",
                     "chr4",
                     "chr5",
                     "chr6",
                     "chr7",
                     "chr8",
                     "chr9",
                     "chr10",
                     "chr11",
                     "chr12",
                     "chr13",
                     "chr14",
                     "chr15",
                     "chr16",
                     "chr17",
                     "chr18",
                     "chr19",
                     "chr20",
                     "chr21",
                     "chr22",
                     "chrX",
                     "chrY",
                     "chrHPV42")

```

# Sample-169
```{r}

# filter specific sample out
manta.calls.169 <- manta.calls %>% filter(sample == 'VD-169') 

# bring data into circlize input format
start.bed <- manta.calls.169 %>% select(chrONE, startONE, endONE, sample)
end.bed <- manta.calls.169 %>% select(chrTWO, startTWO, endTWO, sample)
start.bed$startONE <- start.bed$startONE * 100000
start.bed$endONE <- start.bed$endONE * 100000
start.bed$endONE <- 791700000 - start.bed$endONE
start.bed$startONE <- 791700000 - start.bed$startONE


circos.clear()

# small figure with all integration sites in black
pdf("./plots/Sample-169_circos_all_2x2.pdf", width = 2, height = 2)

circos.initializeWithIdeogram(cytoband = "./data/hg38_genome/cytoBand.txt", chromosome.index = chromosome.list)

# one color for all links
circos.genomicLink(region1 = start.bed, region2 = end.bed, col = "black")

title(main = "Sample-169")

dev.off()

```

